src/GT.Core/Services/Impl/GTListingService.cs

```diff
@@ -43,106 +45,107 @@ public class GTListingService : IGTListingService
 			_experienceLevelRepository = experienceLevelRepository;
 		}
 
-		/// <summary>
-		/// Converts a DTO to entities and updates the database. 
-		/// Requires the signed in users Id for assignment of CreatedBy property.
-		/// </summary>
-		/// <param name="listingDTO"></param>
-		/// <returns>The input DTO with an updated Id.</returns>
-		public async Task<ListingDTO> AddAsync(ListingDTO listingDTO, string signedInUserId)
+		public async Task<ListingDTO?> AddAsync(ListingDTO listingDTO, string signedInUserId)
 		{
-			if (listingDTO is null)
+			try
 			{
-				_logger.LogError($"Attempted to add a null reference to the database.");
-				return null;
-			}
-
-			// Map DTO to entity
-
-			var newListing = await CreateListingEntity(listingDTO, signedInUserId);
-
-			// Update database
-			await _listingRepository.AddAsync(newListing);
-
+				if (listingDTO is null)
+				{
+					_logger.LogWarning($"Attempted to add a null reference to the database.");
+					return null;
+				}
+				if (signedInUserId is null)
+				{
+					_logger.LogWarning($"Attempted to add a new listing with a null reference on signedInUserId.");
+					return null;
+				}
 
-			// Map updates to DTO
-			listingDTO.Id = newListing.Id;
+				var newListing = await CreateListingEntityWithSubEntities(listingDTO, signedInUserId);
+				await _listingRepository.AddAsync(newListing);
 
-			// return the updated ListingDTO
+				// Assigning the updated id to the DTO as it is the only property with a new value.
+				listingDTO.Id = newListing.Id;
+			}
+			catch (Exception e)
+			{
+				_logger.LogError(e.Message);
+				return null;
+			}
 			return listingDTO;
 		}
 
-		public async Task<Listing> CreateListingEntity(ListingDTO listingDTO, string signedInUserId)
+		public async Task<Listing> CreateListingEntityWithSubEntities(ListingDTO listingDTO, string? signedInUserId = null, string? listingId = null)
 		{
-			var newListing = new Listing()
+			var newListingEntity = new Listing();
+
+			newListingEntity.Id = listingId == null ? Guid.NewGuid().ToString() : listingId;
```

Comment:
Could (should) this ever be anything than null? If so, maybe better to overwrite any value to NewGuid()...

Reply:
Will refactor method in up coming PR.